### PR TITLE
urllib3 v2 support

### PR DESCRIPTION
**Pull request by pray**
This PR adds support for urllib3 v2.x, following [urllib3 1.x-2.x migration guide](https://urllib3.readthedocs.io/en/latest/v2-migration-guide.html).

Projects have been adopting urllib3 v2, and in my case it's Airflow 2.x. This requires removing support for Python 2.

### CODE REVIEW

**Comment by TimPansino** [comment]
I'll have to check and see if I have permission to remove Python 2.7 support for this project. Otherwise I'm happy to accept this PR.

There seems to be an issue with Urllib3 v2 on windows, looks like something related to the proxy settings. If you have a chance to see what that's about you can fix it in this PR, or I'll circle back and get it myself later when I have permission to move this forward.

**Comment by thesuperzapper** [comment]
@pray thanks for this, it's really important that `newrelic-telemetry-sdk` is supported in airflow, which currently uses `urllib3==2.2.2` (as of Airflow 2.10.1).

File: src/newrelic_telemetry_sdk/client.py [hunk 1]
```diff
@@ -161,7 +161,6 @@ def __init__(self, license_key, host=None, port=443):
             port=port,
             retries=retries,
             headers=headers,
-            strict=True,
```
Comment (TimPansino) [inline src/newrelic_telemetry_sdk/client.py:164]: Documenting here that this param was Python 2 only, with the removal of Python 2 there's no need to add it anymore.

**Comment by TimPansino** [comment]
Redirecting this to a dev branch so I can figure out the windows issue separately, and fix up the CI.

**Pull request merged**